[Docs] Add teleport.yaml docs for x11 forwarding (#10561)
gravitational/teleport: docs/pages/setup/reference/config.mdx

    # Enables/disables TCP forwarding. Default is 'true'
    port_forwarding: true

>     # When enabled, User's with the "permit_x11_forwarding" role option

Reviewer: ```suggestion
    # When x11.enabled is set to yes, users with the "permit_x11_forwarding" role option
```
This is slightly more than 80 chars, so we'll probably need to reflow this comment

    # will be able to request x11 forwarding sessions with "tsh ssh -X".
    #
    # x11 forwarding will only work if the server has the "xauth" binary installed

>     # and supports unix sockets, e.g. "$TEMP/.X11-unix/X[display_number]".

Reviewer: ```suggestion
    # and supports Unix sockets, e.g. "$TEMP/.X11-unix/X[display_number]."
```

    #

>     # display_offset and max_display can be used to set a specific range

Reviewer: Should we move the descriptions of `display_offset` and `max_display` above the relevant config fields?

    # of x11 displays to use when granting x11 forwarding sessions. This

>     # may be useful on servers with multiple Teleport Nodes or other x11

Reviewer: ```suggestion
    # may be useful on servers with multiple Teleport nodes or other x11
```
I think the docs are moving toward capitalizing "Node" since it's a product name, but this is the only capitalized instance of "node" outside of a link title on this particular page. I've made this suggestion for consistency.

    # services, or on nodes which want to support >1000 concurrent x11
    # forwarding sessions.
    x11: 

>       # "no" by default

Reviewer 2: `"no"` is different than `no` in YAML, as are `"10"` and `10`. I would remove the quotes.

      display_offset: 10
      # max_display can be set to specify the end of the range of X11 displays
      # to use when granting X11 forwarding sessions

>       # isplay_offset + 1000 by default

Reviewer 3: ```suggestion
      # display_offset + 1000 by default
```
